stellargraph/core/graph.py

```diff
@@ -821,7 +821,7 @@ def _unique_type_triples(self, *, return_counts, selector=slice(None)):
 
         return zip(*unique_ets)
 
-    def info(self, show_attributes=True, sample=None, truncate=20):
+    def info(self, show_attributes=None, sample=None, truncate=20):
```

Comment:
if `show_attributes` and `sample` are not used, why are we still keeping it in the `info` definition?

Reply:
Good question: it's not truly needed, but it's helpful for our users. If a user has existing code that has `some_graph.info(sample=20)` or `some_graph.info(False)`, removing the parameters to make this `def info(self, truncate=20)`  will cause their code to break when they upgrade their version of StellarGraph. For example, the first one will have an exception
```python
>>> some_graph.info(sample=20)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: info() got an unexpected keyword argument 'sample'
```
and the second will end up passing `False` to `truncate`, which doesn't give the behaviour they were expecting.

By leaving them here with a warning, the code will continue to work, and the warning will hopefully help them migrate to the better form `some_graph.info()`. After we do a release or two with the deprecations, we can then properly remove them This is similar to how we had a variety of deprecated things in the 0.10 release, and then removed them soon after we did the release in https://github.com/stellargraph/stellargraph/pull/969.